bundles/org.eclipse.ui.workbench/Eclipse UI/org/eclipse/ui/internal/progress/TaskBarProgressManager.java

```diff
@@ -83,46 +80,17 @@ public TaskBarProgressManager(TaskItem taskItem) {
 	 */
 	public void dispose() {
 		ProgressManager.getInstance().removeListener(listener);
-		setAnimated(false);
 		disposeOverlay();
 	}
 
-	private WorkbenchJob getAnimationUpdateJob() {
-		return new WorkbenchJob(ProgressMessages.AnimationManager_AnimationStart) {
+	private WorkbenchJob getUpdateJob() {
+		return new WorkbenchJob(ProgressMessages.ProgressToolItem_Update) {
 
 			@Override
 			public IStatus runInUIThread(IProgressMonitor monitor) {
-				if (isAnimated) {
-					if (!taskItem.isDisposed() && !jobs.isEmpty()) {
-						Job job = jobs.get(0);
-						JobInfo jobInfo = jobInfoMap.get(job);
-						if (job != null && jobInfo != null) {
-							int percentDone = getPercentDone(jobInfo);
-							Optional<TaskInfo> optionalInfo = jobInfo.getTaskInfo();
-							if (percentDone == IProgressMonitor.UNKNOWN
-									|| (optionalInfo.isPresent() && optionalInfo.get().totalWork == IProgressMonitor.UNKNOWN)) {
-								setProgressState(SWT.INDETERMINATE);
-							} else {
-								setProgressState(SWT.NORMAL);
-								if (!taskItem.isDisposed()) {
-									taskItem.setProgress(percentDone);
-								}
-							}
-						} else {
-							setProgressState(SWT.DEFAULT);
-						}
-						updateImage(job);
-					} else {
-						updateImage(null);
-					}
-				} else {
-					setProgressState(SWT.DEFAULT);
-					updateImage(null);
-				}
+				setProgressState(SWT.DEFAULT);
+				updateImage(null);
```

Comment:
What is the purpose of the job now? If it does always same, what's the point to schedule it at all?

Reply:
Who uses it amd wher / for what reason it is scheduled now?

Reply:
Without the job you would not get an moving progress bar.